File: billing-v3/src/test/java/io/appmetrica/analytics/billingv3/impl/ProductTypeParserTest.java
package io.appmetrica.analytics.billingv3.impl;

import com.android.billingclient.api.BillingClient;
import io.appmetrica.analytics.billinginterface.internal.ProductType;
import java.util.Arrays;
import java.util.Collection;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.ParameterizedRobolectricTestRunner;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(ParameterizedRobolectricTestRunner.class)
public class ProductTypeParserTest {

    private final String productType;
    private final ProductType expected;

    public ProductTypeParserTest(final String productType,
                                 final ProductType expected) {
        this.productType = productType;
        this.expected = expected;
    }

    @ParameterizedRobolectricTestRunner.Parameters(name = "{0}")
    public static Collection<Object[]> data() {
        return Arrays.asList(new Object[][]{
                {
                        BillingClient.SkuType.INAPP, ProductType.INAPP
                },
                {
                        BillingClient.SkuType.SUBS, ProductType.SUBS
                },
                {
                        "type", ProductType.UNKNOWN
                },
                {
                        "", ProductType.UNKNOWN
                },
        });
    }

    @Test
    public void testParse() {
        assertThat(ProductTypeParser.parse(productType)).isEqualTo(expected);
    }
}

File: billing-v4/src/test/java/io/appmetrica/analytics/billingv4/impl/ProductTypeParserTest.kt
package io.appmetrica.analytics.billingv4.impl

import com.android.billingclient.api.BillingClient
import io.appmetrica.analytics.billinginterface.internal.ProductType
import io.appmetrica.analytics.billingv4.impl.ProductTypeParser.parse
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.ParameterizedRobolectricTestRunner

@RunWith(ParameterizedRobolectricTestRunner::class)
class ProductTypeParserTest(
    private val productType: String,
    private val expected: ProductType
) {

    companion object {

        @JvmStatic
        @ParameterizedRobolectricTestRunner.Parameters(name = "{0}")
        fun data(): Collection<Array<Any>> {
            return listOf(
                arrayOf(BillingClient.SkuType.INAPP, ProductType.INAPP),
                arrayOf(BillingClient.SkuType.SUBS, ProductType.SUBS),
                arrayOf("type", ProductType.UNKNOWN),
                arrayOf("", ProductType.UNKNOWN)
            )
        }
    }

    @Test
    fun testParse() {
        assertThat(
            parse(
                productType
            )
        ).isEqualTo(expected)
    }
}
